lib/class_kit/value_helper.rb

```diff
@@ -2,76 +2,72 @@ module ClassKit
   class ValueHelper
 
     def self.instance
-      if !class_variable_defined?(:@@instance)
-        class_variable_set(:@@instance, ClassKit::ValueHelper.new)
-      end
-      return @@instance
+      @@instance ||= new
     end
 
-    def parse(type: type, value: value)
-      begin
+    def parse(type, value)
       if type == Time
         if value.is_a?(Time)
-          value = value
+          value
         elsif value.is_a?(Integer) || value.is_a?(Float)
-          value = Time.at(value)
+          Time.at(value)
         else
-          value = Time.parse(value)
+          Time.parse(value)
         end
       elsif type == Date
         if value.is_a?(Date)
-          value = value
+          value
         elsif value.is_a?(Integer)
-          value = Date.at(value)
+          Date.at(value)
         else
-          value = Date.parse(value)
+          Date.parse(value)
         end
       elsif type == DateTime
         if value.is_a?(DateTime)
-          value = value
+          value
         elsif value.is_a?(Integer)
-          value = DateTime.at(value)
+          DateTime.at(value)
         else
-          value = DateTime.parse(value)
+          DateTime.parse(value)
         end
       elsif type == :bool
         if value == true || value == false
-          value = value
+          value
         elsif(/(true|t|yes|y|1)$/i === value.to_s.downcase)
-          value = true
+          true
         elsif (/(false|f|no|n|0)$/i === value.to_s.downcase)
-          value = false
+          false
         elsif value != nil
           raise 'Unable to parse bool'
         end
       elsif type == Integer
-        value = Integer(value)
+        Integer(value)
       elsif type == Float
-        value = Float(value)
-      elsif type == BigDecimal
         Float(value)
-        value = BigDecimal.new(value.to_s)
+      elsif type == BigDecimal
+        if value.is_a?(BigDecimal)
+          value
+        else
+          value = value.to_s
```

Comment:
Please note that not doing `value = Float(value)` here fixes a bug with value that might not be representable well enough with Float.